extensions/vscode/src/utils/window.ts

```diff
@@ -14,3 +14,67 @@ export function showErrorMessageWithTroubleshoot(
     " See [Troubleshooting docs](https://github.com/posit-dev/publisher/blob/main/docs/troubleshooting.md) for help.";
   return window.showErrorMessage(msg, ...items);
 }
+
+export function showInformationMsg(msg: string, ...items: string[]) {
+  return window.showInformationMessage(msg, ...items);
+}
+
+type taskFunc = <T>(p: Progress<T>, t: CancellationToken) => Promise<void>;
+const progressCallbackHandlerFactory =
+  (
+    task: taskFunc,
+    cancellable: boolean = false,
+    onCancel?: () => void,
```

Comment:
Minor suggestion: could we combine the `cancellable` and `onCancel` and rely on the `onCancel` to determine if the handler is cancellable? If `onCancel` was passed it is cancellable, if not it isn't.